### PR TITLE
Feature/fix evaluator in layer v2

### PR DESCRIPTION
**Pull request by reyoung**
Related [issue](https://github.com/PaddlePaddle/book/issues/265)

### CODE REVIEW

File: python/paddle/v2/config_base.py [hunk 1]
```diff
@@ -65,13 +65,27 @@ class Layer(object):
     def __init__(self, name=None, parent_layers=None):
         assert isinstance(parent_layers, dict)
         self.name = name
-        self.__contex__ = {}
+        self.__context__ = {}
```
Comment (reyoung) [inline python/paddle/v2/config_base.py]: Here is a typo before. Changed by the way.

File: python/paddle/v2/config_base.py [hunk 2]
```diff
@@ -129,9 +158,20 @@ def __init__(self, **kwargs):
             parent_layers = dict()
             other_kwargs = dict()
             for pname in parent_names:
-                if kwargs.has_key(pname):
+                if pname in kwargs:
```
Comment (reyoung) [inline python/paddle/v2/config_base.py]: More pythonic

File: python/paddle/v2/evaluators.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.trainer_config_helpers.evaluators as evs
+import inspect
+from config_base import __convert_to_v2__
+
+__all__ = []
```
Comment (qingqing01) [inline python/paddle/v2/evaluators.py]: 和layer单复数保持一致，evaluators是不是也用evaluator好些？

File: python/paddle/v2/tests/test_layer.py [hunk 4]
```diff
@@ -262,5 +263,20 @@ def test_vgg(self):
...
+    def test_evaluator(self):
+        img = layer.data(name='pixel', type=data_type.dense_vector(784))
+        output = layer.fc(input=img,
+                          size=10,
+                          act=activation.Softmax(),
+                          name='fc_here')
+        lbl = layer.data(name='label', type=data_type.integer_value(10))
+        cost = layer.cross_entropy_cost(input=output, label=lbl)
+
+        evaluators.classification_error_evaluator(input=output, label=lbl)
```
Comment (qingqing01) [inline python/paddle/v2/tests/test_layer.py]: evaluators.classification_error_evaluator  =>  evaluator.classification_error  ?

File: python/paddle/v2/evaluators.py (hunk 3 above)
Comment (reyoung) [inline python/paddle/v2/evaluators.py]: Done.

File: python/paddle/v2/tests/test_layer.py (hunk 4 above)
Comment (reyoung) [inline python/paddle/v2/tests/test_layer.py]: Done.

File: python/paddle/v2/evaluators.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def initialize():
+    for __ev_name__ in filter(lambda x: x.endswith('_evaluator'), evs.__all__):
+        __ev__ = getattr(evs, __ev_name__)
+        if hasattr(__ev__, 'argspec'):
+            argspec = __ev__.argspec
+        else:
+            argspec = inspect.getargspec(__ev__)
+        parent_names = filter(lambda x: x in ['input', 'label'], argspec.args)
```
Comment (reyoung) [inline python/paddle/v2/evaluators.py]: Done.

**Comment by jacquesqiao** [comment]
LGTM

**Pull request merged**